build/tasks/AddMetapackageReferences.cs

```diff
@@ -81,7 +100,14 @@ public override bool Execute()
             {
                 var packageName = package.ItemSpec;
                 var packageVersion = package.GetMetadata("Version");
-                var packageVersionValue = LockToExactVersions ? $"[{packageVersion}]" : packageVersion;
+
+                if (string.IsNullOrEmpty(packageVersion))
```

Comment:
Why not just put this in the GetDependencyVersion method?

Reply:
The behavior I wanted was for this task to log an error for *all* packages missing version info -- not just the first one it encounters -- and then fail. This was easier to implement by putting the check in this loop and using `continue;` instead of returning early or throwing.

Reply:
Logging the error here versus in the method doesn't make a difference though right? I don't think logging an error prevents the rest of the logic in the loop from executing. Regardless, I don't mind keeping it here either so don't let this hold up the PR.